test/integration/targets/azure_rm_cosmosdbaccount/tasks/main.yml

```diff
@@ -0,0 +1,107 @@
+- name: Prepare random number
+  set_fact:
+    dbname: "cosmos{{ resource_group | hash('md5') | truncate(7, True, '') }}{{ 1000 | random }}"
+  run_once: yes
+
+- name: Create instance of Database Account -- check mode
+  azure_rm_cosmosdbaccount:
+    resource_group: "{{ resource_group }}"
+    name: "{{ dbname }}"
+    location: eastus
+    geo_rep_locations:
+      - name: eastus
+        failover_priority: 0
+    database_account_offer_type: Standard
+  check_mode: yes
+  register: output
+- name: Assert the resource instance is well created
+  assert:
+    that:
+      - output.changed
+
+- name: Create instance of Database Account
+  azure_rm_cosmosdbaccount:
+    resource_group: "{{ resource_group }}"
+    name: "{{ dbname }}"
+    location: eastus
+    geo_rep_locations:
+      - name: eastus
+        failover_priority: 0
+      - name: westus
+        failover_priority: 1
+    database_account_offer_type: Standard
+  register: output
+- name: Assert the resource instance is well created
+  assert:
+    that:
+      - output.changed
+
+- name: Create again instance of Database Account
+  azure_rm_cosmosdbaccount:
+    resource_group: "{{ resource_group }}"
+    name: "{{ dbname }}"
+    location: eastus
+    geo_rep_locations:
+      - name: westus
+        failover_priority: 1
+      - name: eastus
+        failover_priority: 0
+    database_account_offer_type: Standard
+  register: output
+- name: Assert the state has not changed
+  assert:
+    that:
+      - output.changed == false
+
+- name: Create again instance of Database Account -- change something
+  azure_rm_cosmosdbaccount:
+    resource_group: "{{ resource_group }}"
+    name: "{{ dbname }}"
+    location: eastus
+    geo_rep_locations:
+      - name: westus
+        failover_priority: 1
+      - name: eastus
+        failover_priority: 0
+    database_account_offer_type: Standard
+    enable_automatic_failover: yes
+  register: output
+- name: Assert the state has not changed
+  assert:
+    that:
+      - output.changed
+
+- name: Delete instance of Database Account -- check mode
+  azure_rm_cosmosdbaccount:
+    resource_group: "{{ resource_group }}"
+    name: "{{ dbname }}"
+    state: absent
+  check_mode: yes
+  register: output
+- name: Assert the state has changed
+  assert:
+    that:
+      - output.changed
+
+- name: Delete instance of Database Account
+  azure_rm_cosmosdbaccount:
+    resource_group: "{{ resource_group }}"
+    name: "{{ dbname }}"
+    state: absent
+  register: output
+- name: Assert the state has changed
+  assert:
+    that:
+      - output.changed
+
+# currently disabled as there's a bug in SDK / Service
```

Comment:
Does this mean that `state: absent` is not idempotent, or is there a race condition or something specific to the test?

Reply:
the problem is that SDK actually returns before actual Cosmos DB instance is completely deleted.
so trying to delete again actually fails.
i have added code to periodically call GET until instance is gone, but we decided that it's a hack and we shouldn't do that.